Off by one when reading connections from pdb?

There appear to be a few issues when reading bonds from PDB files.

In the function _parsePDBLines, In this situation

```
        elif not onlycoords and (startswith == 'TER   ' or
            startswith.strip() == 'TER'):
            termini[acount - 1] = True
            acount += 1   # This line is necessary
```
The lack of the increment causes descrepancies in the Atom index in the AtomGroup and the Atom index in the PDB (this is because TER entries increment the atom field)

Secondly, The mechanics of the other parts of the code seems to suggest that the Atom index is 0 starting, and it is this index that is used with the bonds. This means that in the preceding lines

```
        elif startswith == 'CONECT':
            if bonds is not None:
                atom_serial = line[6:11]
                bonded1_serial = line[11:16]
                bonds.append([int(atom_serial)-1, int(bonded1_serial)-1])  # -1 necessary
                
                bonded2_serial = line[16:21]
                if len(bonded2_serial.strip()):
                    bonds.append([int(atom_serial)-1, int(bonded2_serial)-1])  # -1 necessary

                bonded3_serial = line[21:26]
                if len(bonded3_serial.strip()):
                    bonds.append([int(atom_serial)-1, int(bonded3_serial)-1])  # -1 necessary
                    
                bonded4_serial = line[27:31]
                if len(bonded4_serial.strip()):
                    bonds.append([int(atom_serial)-1, int(bonded4_serial)-1])  # -1 necessary
```

the -1 is necessary. Otherwise causing several off-by-one errors.

Actually, this is introducing other problems, which the tests have picked up on.

For example, we now have extra entries in the AtomGroup that are picked up in the hierarchical view test.

The original AtomGroup has 9392 atoms like the object in PyMOL and the chains in the HierView have the right numbers of atoms too.
```
james@DESKTOP-QL82H8Q MINGW64 ~/code/ProDy (prody-master)
$ ipython
Python 3.8.6 | packaged by conda-forge | (default, Nov 27 2020, 18:58:29) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.20.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from prody import *

In [2]: ag = parsePDB('3h5v', bonds=True, compressed=False)
@> PDB file is found in working directory (3h5v.pdb).
@> 9392 atoms and 1 coordinate set(s) were parsed in 0.17s.
@> Secondary structures were assigned to 693 residues.

In [3]: ag
Out[3]: <AtomGroup: 3h5v (9392 atoms)>

In [4]: list(ag.getHierView())
Out[4]:
[<Chain: A from 3h5v (605 residues, 3225 atoms)>,
 <Chain: B from 3h5v (494 residues, 3114 atoms)>,
 <Chain: C from 3h5v (394 residues, 2997 atoms)>,
 <Chain: D from 3h5v (2 residues, 28 atoms)>,
 <Chain: E from 3h5v (2 residues, 28 atoms)>]
```

After the fix, we have three extra atoms that don't belong to any chain and are defined together as a residue:
```
In [1]: from prody import *

In [2]: ag = parsePDB('3h5v', bonds=True, compressed=False)
@> PDB file is found in working directory (3h5v.pdb).
@> 9395 atoms and 1 coordinate set(s) were parsed in 0.14s.
@> Secondary structures were assigned to 693 residues.

In [3]: ag
Out[3]: <AtomGroup: 3h5v (9395 atoms)>

In [4]: list(ag.getHierView())
Out[4]:
[<Chain: A from 3h5v (605 residues, 3225 atoms)>,
 <Chain:  from 3h5v (1 residues, 3 atoms)>,
 <Chain: B from 3h5v (494 residues, 3114 atoms)>,
 <Chain: C from 3h5v (394 residues, 2997 atoms)>,
 <Chain: D from 3h5v (2 residues, 28 atoms)>,
 <Chain: E from 3h5v (2 residues, 28 atoms)>]
```

Likewise, the simple test system https://github.com/prody/ProDy/blob/master/prody/tests/datafiles/pdbRTER.pdb has 2 amino acid residues and 7 waters, giving a total of 9, but the test fails as it it assigns the 8 TER entries as atoms belonging to a 10th residue:
```
________________________ TestTerRecord.testNumResidues _________________________

self = <prody.tests.atomic.test_hierview.TestTerRecord testMethod=testNumResidues>

    def testNumResidues(self):
    
>       self.assertEqual(RTER.getHierView().numResidues(), 9)
E       AssertionError: 10 != 9
```

--- a/prody/proteins/pdbfile.py
+++ b/prody/proteins/pdbfile.py
@@ -737,23 +737,24 @@ def _parsePDBLines(atomgroup, lines, split, model, chain, subset,
             if bonds is not None:
                 atom_serial = line[6:11]
                 bonded1_serial = line[11:16]
-                bonds.append([int(atom_serial), int(bonded1_serial)])
+                bonds.append([int(atom_serial)-1, int(bonded1_serial)-1])
                 
                 bonded2_serial = line[16:21]
                 if len(bonded2_serial.strip()):
-                    bonds.append([int(atom_serial), int(bonded2_serial)])
+                    bonds.append([int(atom_serial)-1, int(bonded2_serial)-1])
 
                 bonded3_serial = line[21:26]
                 if len(bonded3_serial.strip()):
-                    bonds.append([int(atom_serial), int(bonded3_serial)])
+                    bonds.append([int(atom_serial)-1, int(bonded3_serial)-1])
                     
                 bonded4_serial = line[27:31]
                 if len(bonded4_serial.strip()):
-                    bonds.append([int(atom_serial), int(bonded4_serial)])
+                    bonds.append([int(atom_serial)-1, int(bonded4_serial)-1])
 
         elif not onlycoords and (startswith == 'TER   ' or
             startswith.strip() == 'TER'):
             termini[acount - 1] = True
+            acount += 1   # This line is necessary
         elif startswith == 'ENDMDL' or startswith[:3] == 'END':
             if acount == 0:
                 # If there is no atom record between ENDMDL & END skip to next

--- a/prody/proteins/pdbfile.py
+++ b/prody/proteins/pdbfile.py
@@ -422,6 +422,8 @@ def _parsePDBLines(atomgroup, lines, split, model, chain, subset,
     format = format.upper()
     isPDB = format == 'PDB'
 
+    num_ters = 0
+
     if subset:
         if subset == 'ca':
             subset = set(('CA',))
@@ -737,24 +739,24 @@ def _parsePDBLines(atomgroup, lines, split, model, chain, subset,
             if bonds is not None:
                 atom_serial = line[6:11]
                 bonded1_serial = line[11:16]
-                bonds.append([int(atom_serial)-1, int(bonded1_serial)-1])
+                bonds.append([int(atom_serial)-1-num_ters, int(bonded1_serial)-1-num_ters])
                 
                 bonded2_serial = line[16:21]
                 if len(bonded2_serial.strip()):
-                    bonds.append([int(atom_serial)-1, int(bonded2_serial)-1])
+                    bonds.append([int(atom_serial)-1-num_ters, int(bonded2_serial)-1-num_ters])
 
                 bonded3_serial = line[21:26]
                 if len(bonded3_serial.strip()):
-                    bonds.append([int(atom_serial)-1, int(bonded3_serial)-1])
+                    bonds.append([int(atom_serial)-1-num_ters, int(bonded3_serial)-1-num_ters])
                     
                 bonded4_serial = line[27:31]
                 if len(bonded4_serial.strip()):
-                    bonds.append([int(atom_serial)-1, int(bonded4_serial)-1])
+                    bonds.append([int(atom_serial)-1-num_ters, int(bonded4_serial)-1-num_ters])
 
         elif not onlycoords and (startswith == 'TER   ' or
             startswith.strip() == 'TER'):
             termini[acount - 1] = True
-            acount += 1   # This line is necessary
+            num_ters += 1
         elif startswith == 'ENDMDL' or startswith[:3] == 'END':
             if acount == 0:
                 # If there is no atom record between ENDMDL & END skip to next

